executor/executor_xapi.go

```diff
@@ -196,12 +206,14 @@ func (e *XSelectIndexExec) Next() (*Row, error) {
 		if e.indexPlan.LimitCount != nil {
 			limitCount = *e.indexPlan.LimitCount
 		}
+		concurrency := 2
 		if e.indexPlan.NoLimit {
-			e.runTableTasks(len(e.tasks))
+			concurrency = len(e.tasks)
 		} else if limitCount > int64(BaseLookupTableTaskSize) {
-			concurentCount := limitCount/int64(BaseLookupTableTaskSize) + 1
-			e.runTableTasks(int(concurentCount))
+			concurrency = int(limitCount/int64(BaseLookupTableTaskSize) + 1)
```

Comment:
if limitCount > int64(BaseLookupTableTaskSize), "limitCount/int64(BaseLookupTableTaskSize" should not be less than 1. So concurrency should not be less than 2.